### PR TITLE
Block Info Test

### PR DESCRIPTION
**Pull request by winsvega**
Should block.info be equal at state and the blockchain?

### CODE REVIEW

File: test/libethereum/blockchain.cpp [hunk 1]
```diff
@@ -795,4 +795,45 @@ BOOST_AUTO_TEST_CASE(userDefinedFile)
...
+	bytes blockRLP = dev::test::importByteArray(blockRLPstr);
+	BlockInfo block(blockRLP);
+	trueBc.import(blockRLP, trueState.db());
+	trueState.sync(trueBc);
+
+	if (trueState.info() != trueBc.info())
```
Comment (subtly) [inline test/libethereum/blockchain.cpp]: Can BOOST_CHECK or BOOST_REQUIRE be used here?

**Comment by subtly** [comment]
What does this test?

File: test/libethereum/blockchain.cpp (hunk 1 above)
Comment (winsvega) [inline test/libethereum/blockchain.cpp]: No. Too much output with '<<'

**Comment by winsvega** [comment]
https://github.com/ethereum/cpp-ethereum/issues/1940  
I suppose this (info()) values should be equal. I might be wrong. 
However it is not equal. I wrote this test to see that it is not equal for sure. So it might be an issue.

**Comment by CJentzsch** [comment]
It should be equal, please dig a little deeper. Output the complete BlockInfo for both to see where's the difference.
Also do a try catch around the sync and the block import, could be that one of them failed.

**Comment by winsvega** [comment]
I found. 
https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/State.cpp#L340

The state is resetting it's current block to be the next one. So it returns i+1 block after getting synced with the blockchain. May be this is how it supposed to be. Just confusing.

**Comment by CJentzsch** [comment]
If so, than the test is wrong.

**Comment by winsvega** [comment]
I still think state and blockchain interfaces should be refactored.

**Pull request closed without merging**